Feature/data submission workflow

fiedosiukr opened the pull request:
This PR creates a new section that allows authors to upload additional approvals. In order to submit a project with access policy set to credentialed, either ethical approval or an explanation must be included.

![image](https://user-images.githubusercontent.com/44640544/140751065-b922b745-0fa7-483d-b4bf-222e51cfed4a.png)

Commit:
Add approvals section.

Commit:
Add new editor question.

fiedosiukr committed:
Move approvals section to admin console.

fiedosiukr committed:
Create custom file input widget.

fiedosiukr committed:
Copy approvals when creating a new version.

Commit:
Fix styles.

Commit:
Change field names, fix code styles.

Commit:
Add mising underscore.

Commit:
Rename approvals to ethics.

Commit:
Add dynamic formset.

fiedosiukr committed:
Delete documents when deleting objects.

Commit:
Rename models.

alistairewj commented on physionet-django/project/forms.py:
```diff
@@ -1041,3 +1047,52 @@ def save(self):
+                '* A statement regarding ethical concerns for the work. '
+                'This statement will be published with the resource, '
+                'and typically describes formal approvals acquired for '
+                'the creation of the resource, and/or ethical considerations '
```
change to:

"the creation of the resource (such as a review by an ethics board) ..."

fiedosiukr replied:
🆗

alistairewj commented on physionet-django/project/modelcomponents/activeproject.py:
```diff
@@ -563,7 +575,7 @@ def publish(self, slug=None, make_zip=True, title=None):
                 return published_project
 
-        except:
+        except Exception:
```
you may run into trouble for this one based off previous conversations!

fiedosiukr replied:
Yeah, I know, I forgot to push the final changes that I made. I will do it tomorrow.

alistairewj commented on physionet-django/project/modelcomponents/submission.py:
```diff
@@ -32,7 +62,8 @@ class EditLog(models.Model):
         'no_phi': 'Is the project free of protected health information?',
-        'data_machine_readable': 'Are all files machine-readable?'
+        'data_machine_readable': 'Are all files machine-readable?',
+        'ethics_included': 'Does the submission contain the necessary approvals?',
```
change to:

'Does the submission contain an appropriate statement regarding ethical concerns, and if necessary, supporting documentation?'

fiedosiukr replied:
🆗

alistairewj commented on physionet-django/project/modelcomponents/activeproject.py:
```diff
@@ -310,6 +313,11 @@ def check_integrity(self):
                 self.integrity_errors.append('Missing required field: {0}'.format(l))
 
+        # Approvals
+        if self.access_policy == 2:
```
you for sure remember this from the meeting but just noting it anyway to remove the access policy restriction

fiedosiukr replied:
🆗

alistairewj commented on physionet-django/project/forms.py:
```diff
@@ -1041,3 +1047,52 @@ def save(self):
+        super().__init__(*args, **kwargs)
+        self.help_text = (
+            'Upload supporting documents. For more information on '
+            'what documents are expected, see the [author guidance page].'
```
can the [author guidance page] be changed to a hyperlink to the author guidance page? currently it's here: https://physionet.org/about/publish/#guidelines

also, add this sentence at the end: "These documents reviewed by the editorial staff, and are not published with the project."

fiedosiukr replied:
🆗

alistairewj reviewed (changes requested):
Here are the suggestions I promised in the call. Also, can you add the following to the publish section of the author guidelines, below the "Version" bullet point in the "Creating the project metadata" header:

* **Ethics**: Provide a statement regarding the ethical concerns of the resource. This statement should include any formal ethical approval acquired in the curation of the resource, as well as any concerns individuals should be aware of when using the resource. The ethics statement will be published with the project description. Include an approval letter from the respective ethics board and the associated study protocol if ethics approval was acquired (these documents will be reviewed by the editor, but they will not be published with the resource).

Commit:
Add descriptions.

Commit:
Move ethics section above files section.

Commit:
Add DocumentType fixtures.

Commit:
Add missing quote.

Commit:
Serve files for local file system.

Commit:
Add ethics_statement to the fixtures file.

Commit:
Add newline.

Commit:
Apply code review suggestions.

Commit:
Fix migrations.

Commit:
Improve supporting document submission.

Commit:
Fix migrations.

Commit:
Format the code.

tompollard committed:
Leave ethics section empty for published projects.

It could be misleading to populate previously published projects with this statement. Instead, please leave this blank. The project templates will omit the ethics section when empty.

fiedosiukr committed:
Merge pull request #26 from MIT-LCP/upsidelab-feature/data-submission-workflow

Leave ethics section empty for published projects.
